unittests/Dialect/SPIRV/Runtime/RuntimeTests.cpp

```diff
@@ -0,0 +1,150 @@
+//===- RuntimeTest.cpp - SPIR-V Runtime Tests -----------------------------===//
+//
+// Copyright 2019 The MLIR Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =============================================================================
+//
+// This file is to test Vulkan runtime by public function which consumes
+// spirv::ModuleOp. This file will be removed in the future.
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Dialect/SPIRV/SPIRVBinaryUtils.h"
+#include "mlir/IR/Location.h"
+#include "mlir/IR/MLIRContext.h"
+#include "mlir/IR/Module.h"
+#include "mlir/Parser.h"
+#include "mlir/Support/FileUtilities.h"
+#include "mlir/Support/LogicalResult.h"
+#include "mlir/Support/VulkanRuntime.h"
+#include "llvm/ADT/DenseMap.h"
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/Support/SourceMgr.h"
+#include "llvm/Support/ToolOutputFile.h"
+
+#include "gmock/gmock.h"
+#include <string>
+#include <cstdlib>
+#include <ctime>
+
+using namespace mlir;
+using namespace llvm;
+
+class RuntimeTest : public ::testing::Test {
+protected:
+  RuntimeTest() {}
+
+  LogicalResult parseAndRunOnModule(llvm::StringRef fileName,
+                                    NumWorkGroups numWorkGroups) {
+    std::string errorMessage;
+    auto inputFile = openInputFile(fileName, &errorMessage);
+    if (!inputFile) {
+      llvm::errs() << errorMessage << "\n";
+      return failure();
+    }
+
+    SourceMgr sourceMgr;
+    sourceMgr.AddNewSourceBuffer(std::move(inputFile), SMLoc());
+
+    MLIRContext context;
+    OwningModuleRef moduleRef(parseSourceFile(sourceMgr, &context));
+    if (!moduleRef) {
+      llvm::errs() << "\ncannot open the file" << '\n';
+      return failure();
+    }
+
+    if (failed(runOnModule(moduleRef.get(), vars, numWorkGroups))) {
+      return failure();
+    }
+
+    return success();
+  }
+
+  void createResourceVarFloat(uint32_t id, uint32_t elementCount) {
+    std::srand(unsigned(std::time(0)));
+    float *ptr = new float[elementCount];
```

Comment:
@antiagainst can you please specify, do I unrerstand you right, you want to update VulkanHostMemoryBuffe to hold unique ptr instead pointer to void*?

Reply:
No. Actually it's fine to have VulkanHostMemoryBuffer to hold raw pointer. I was just saying that we should wrap these raw pointers in `unique_ptr` to avoid potential leaks. For example, you can let this function to return the `unique_ptr` and assign it to a local variable in the test so we don't need to explicitly call `delete` later.